cmd/fetchd/cmd/genasiupgrade.go

```diff
@@ -163,6 +178,74 @@ func ASIGenesisUpgradeCmd(defaultNodeHome string) *cobra.Command {
 	return cmd
 }
 
+func ASIGenesisUpgradeUpdateMobixStakingContract(jsonData map[string]interface{}, networkInfo NetworkConfig) {
+	contracts := jsonData["wasm"].(map[string]interface{})["contracts"].([]interface{})
+	MobixStakingContractAddress := networkInfo.Contracts.MobixStaking.Addr
+
+	re := regexp.MustCompile(fmt.Sprintf(`%s%s1([%s]{%d})$`, OldAddrPrefix, "", Bech32Chars, AddrDataLength+AddrChecksumLength))
```

Comment:
```suggestion
	re := regexp.MustCompile(fmt.Sprintf(`%s%s1([%s]{%d,%d})$`, OldAddrPrefix, "", Bech32Chars, 32+AddrChecksumLength, 52+AddrChecksumLength))
```
or, probably even better:
```suggestion
	re := regexp.MustCompile(fmt.Sprintf(`%s%s1([%s]{%d,%d})$`, OldAddrPrefix, "", Bech32Chars, AddrChecksumLength, 52+AddrChecksumLength))
```

Reply:
I would use leven higher value then 52 (which supposed to be for contract address, and at the moment it looks like it is largest address length we know of), for example at least 60, or even 100.